subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ConfigurationCacheState.kt

```diff
@@ -208,6 +208,9 @@ class ConfigurationCacheState(
             writeRequiredBuildServicesOf(gradle, buildTreeState)
```

Comment:
I am trying to understand the reason between the two unchanged lines above. Could you provide some explanation? Without that context, I find confusing that `writeProjectStates()` does not call `writeProjectState()`, but `writeRelevantProjects()` does.

Reply:
Good catch! I didn't realize there was a function with a similar name already there. `writeRelevantProjects` writes just the state required for recreating the relevant project structure when loading from the cache while `writeProjectStates` writes the _configured_ project state. `writeProjectState` should be renamed `writeProjectRegistration` or similar.

Reply:
I've renamed the relevant (no pun intended) methods.